sdl_viewer/src/terrain_drawer/layer.rs

```diff
@@ -218,10 +219,18 @@ impl GridCoordinateFrame {
     /// Returns the terrain pos (i.e. the coordinate of the lower corner of the terrain) for
     /// a given camera position (in the world coordinate system).
     fn terrain_pos_for_camera_pos(&self, world_pos: Point3<f64>) -> Vector2<i64> {
-        let local_pos = &self.terrain_from_world * &world_pos;
+        let local_pos = self.terrain_from_world * world_pos;
         let x = ((local_pos.x - self.u_origin.value.x) / self.u_resolution_m.value).floor();
         let y = ((local_pos.y - self.u_origin.value.y) / self.u_resolution_m.value).floor();
-        (Vector2::new(x, y)).cast().unwrap() - self.texture_half_extent
+        assert!(
```

Comment:
Would it be possible to reuse the asserts of `convert_terrain_pos`?

Reply:
No, I don't think so. One is comparing floats and one ints, and they have different safe ranges.